### PR TITLE
add create mint entitlement json sample

### PR DESCRIPTION
**Pull request by elv-todd**
for https://github.com/qluvio/elv-apps-projects/issues/243

```
$ node samples/GenMintEntitlement.js iten4TXq2en3qtu3JREnE5tSLRf9zLod iq__2dXeKyUVhpcsd1RM6xaC1jdeZpyr C9Zct19CoEAZYWug9tyavX 1 nonce_6f9f53ecc09a7e223cf7d47f pid_e852572c6e84626892da049
SIGNER 0x761f45287ea364db6b216bd655910430afa3e839
ENTITLEMENT TO SIGN {"tenant_id":"iten4TXq2en3qtu3JREnE5tSLRf9zLod","marketplace_id":"iq__2dXeKyUVhpcsd1RM6xaC1jdeZpyr","items":[{"sku":"C9Zct19CoEAZYWug9tyavX","amount":1}],"nonce":"nonce_6f9f53ecc09a7e223cf7d47f","purchase_id":"pid_e852572c6e84626892da049"}
ENTITLEMENT {
  tenant_id: 'iten4TXq2en3qtu3JREnE5tSLRf9zLod',
  marketplace_id: 'iq__2dXeKyUVhpcsd1RM6xaC1jdeZpyr',
  items: [ { sku: 'C9Zct19CoEAZYWug9tyavX', amount: 1 } ],
  nonce: 'nonce_6f9f53ecc09a7e223cf7d47f',
  purchase_id: 'pid_e852572c6e84626892da049'
}
ENTITLEMENT_SIGNATURE ES256K_7uAQCp6nUQDiyHUSUa4AHV91rTpgoUsRQ4pPrjZpY9PZ3Ww21ZFaF4q6zXKP2TFQajbvUMS38PGeyA6XERwpcasbh
ALL DONE
```

### CODE REVIEW

File: samples/GenMintEntitlement.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ */
+const Entitlement = async({tenant, marketplaceObjectId, sku, amount, nonce, purchaseId}) => {
+  const message = {
+    tenant_id: tenant,
+    marketplace_id: marketplaceObjectId,
+    items: [ { sku: sku, amount: amount } ],
+    nonce: nonce,
+    purchase_id: purchaseId,
+  };
+
```
Comment (elv-preethi) [inline samples/GenMintEntitlement.js:33]: should it be `personal_sign` message?

```
message = typeof message === "object" ? JSON.stringify(message) : message;
message = Ethers.utils.keccak256(Buffer.from(`\x19Ethereum Signed Message:\n${message.length}${message}`, "utf-8"));
```

Comment (elv-todd) [reply samples/GenMintEntitlement.js:33]: nah we want it to use the existing client function with as little dressing as possible.  if the client lib wants to do that internally in some client.SignZXC(), that's fine tho.

**Comment by elv-todd** [comment]
updated to use new sig format:

```
$ node samples/GenMintEntitlement.js
SIGNER 0x761f45287ea364db6b216bd655910430afa3e839
ENTITLEMENT {
  tenant_id: 'iten4TXq2en3qtu3JREnE5tSLRf9zLod',
  marketplace_id: 'iq__2dXeKyUVhpcsd1RM6xaC1jdeZpyr',
  items: [ { sku: 'C9Zct19CoEAZYWug9tyavX', amount: 1 } ],
  nonce: 'nonce_6f9f53ecc09a7e223cf7d47f',
  purchase_id: 'pid_e852572c6e84626892da049a'
}
ENTITLEMENT_SIGNATURE mje_2LDb4upTtBDeQdcAt2oBvya1338s3WVdrzCQZFvXo1YLiAmWvh44xAZcdpJyzhtcuG4s6PNXgcBYbfrbAPxNK3bYEUYkv9Pp6uHGvvP8V8f5bPDk6qMYvYCvH94AsrxhqXY3Gsn98uTNioe6711ZFpQTBjDukL4Xt7mzvN88NvxU5oWCV2W1vzn1VPnPrVZZ6TSkv4fLTHj1bJfL5NWSgDBAcQYAf9pdNgiJX9reYQGKceQk5LViqNfjSBXhzd4rgbvBJMD8Uua8mXo1zqxCoyZrR5UhfvpsNX3oSd9P8Vw5zVDNX3s9JpPR8edZ73EbZixoiYf5h6bJfZggHJDxcgkzzD5Unhcbz3MxA7XS6yZ32A1dsXfWtc1m2AQU1vLEmoyr9oQG9JCURznZhFtHGR4qhieBnWMZN
ALL DONE
```

File: samples/GenMintEntitlement.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+const crypto = require("crypto");
+const { ethers } = require('ethers');
+const bs58 = require("bs58");
+
+let client;
+
+/* Sample configuration -- demov3 network */
+let tenant = "iten4TXq2en3qtu3JREnE5tSLRf9zLod"; // paladin
+let marketplaceObjectId = "iq__2dXeKyUVhpcsd1RM6xaC1jdeZpyr"; // A Place for Goats
+let sku = "C9Zct19CoEAZYWug9tyavX"; // Goat Pack One
```
Comment (elv-serban) [inline samples/GenMintEntitlement.js:11]: I would remove the comments referencing the names of test tenants and marketplaces.  It's fine to leave the IDs as examples.

**Review by elv-serban — APPROVED** [review]
I think this is fine to merge and then modify to use a more official elv-client-js CreateSignedMessageJSON once we have one

**Pull request merged**